Adding TLS support to db_redis (also password support) and ndb_redis

joelbax opened the pull request:
<!-- Kamailio Pull Request Template -->

<!--
IMPORTANT:
  - for detailed contributing guidelines, read:
    https://github.com/kamailio/kamailio/blob/master/.github/CONTRIBUTING.md
  - pull requests must be done to master branch, unless they are backports
    of fixes from master branch to a stable branch
  - backports to stable branches must be done with 'git cherry-pick -x ...'
  - code is contributed under BSD for core and main components (tm, sl, auth, tls)
  - code is contributed GPLv2 or a compatible license for the other components
  - GPL code is contributed with OpenSSL licensing exception
-->

#### Pre-Submission Checklist
<!-- Go over all points below, and after creating the PR, tick all the checkboxes that apply -->
<!-- All points should be verified, otherwise, read the CONTRIBUTING guidelines from above-->
<!-- If you're unsure about any of these, don't hesitate to ask on sr-dev mailing list -->
- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:
<!-- Go over all points below, and after creating the PR, tick the checkboxes that apply -->
- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description
A couple of commits into db_redis and ndb_redis adding TLS support and also password support to db_redis.

This mainly includes checking if proper parameter is provided (for ndb_redis is `tls` option in the DB URL and, for db_redis, a new `opt_tls` parameter) and creates a temporary SSL context that is used to initialise the redis context.

Also added `ca_path` parameter to both modules to be able to define a valid folder containing the root certificates used to validate TLS' certificate chain.

db_redis is also updated with a `db_pass` parameter to provide a DB access password.

TLS support is automatically enabled by checking libhiredis_ssl.so existence in each Makefile and defining a `WITH_SSL` flag that enables all the corresponding code lines.

joelbax commented:
@henningw  This is the new PR for adding TLS support to Redis. Once again, I'm sorry for all the nuisance.
@oej Also fyi (as you commented on the original PR).
Thank you very much in advance!

miconda commented on src/modules/db_redis/db_redis_mod.c:
```diff
@@ -29,6 +29,12 @@
+int db_redis_opt_tls = 0;
+char *ca_path = 0;
+#endif
+char *db_pass = 0;
```
Add module name prefix to the global variables ca_path and db_pass, like it is db_redis_opt_tls.  The are also declared with extern and being prefixed reduce the chances of conflicts with similar variables (e.g., ca_path can be quite common with some libs).

joelbax replied:
Very good point! Thank you very much. 

Done.

henningw commented:
Or maybe just as other modules are doing (like ndb_redis), assume a few default paths if there is no pkg-config available.

joelbax commented:
@miconda I've added logic to check gcc search paths for libraries in case `pkg-config` returns nothing.

I've also fixed som formatting issues but  the check-format task seems to complain about some lines but , I'm really sorry but, I can't see the difference between the wrong line and the new one.

miconda commented:
Thanks!

miconda commented:
@joelbax: I merged and then updated a bit the find command in the Makefile for n/db_redis modules to silent some warning about missing directories and use `-name` for finding file expression. Now compiles fine on ubuntu 22.04, before I got errors. Maybe you can test on your system to be sure it didn't break something else.
